magento/pwa-studio
# [feature] Add products to wishlist from peregrine gallery

## Description

In #3105 the `Gallery` and `GalleryItem` components were augmented with logic and functionality to:

* initialize a local, cached list of skus within a customer's wishlists and
* render a button (the heart) for adding an item to a list
* render that button as filled or not, based on whether the product sku is in the local array

This PR takes the majority of that logic, specifically the first bullet above, and moves it into a re-useable hook, `useCustomerWishlistSkus` that can be run in a component to "prep" the cache array for comparison by some child component such as the [GalleryButton](https://github.com/magento/pwa-studio/blob/pwa-1672/add-to-wishlist-from-cms/packages/peregrine/lib/talons/Wishlist/GalleryButton/helpers/useSingleWishlist.js#L25-L27).

Because of how Pagebuilder's `Products` component was implemented (it renders either a "SlickSlider" or venia's `Gallery` component) I didn't want to call the shared hook directly as that could potentially result in duplicate calls (once in `Products` and once in `Gallery`). Instead, there is a new `Carousel` component which calls this new shared hook and does a minimal amount of work to fetch the `storeConfig` to pass to the `GalleryItems` it renders.

## Potential breaking changes

If this PR is released with #3105 then it is not breaking, but if it goes out in a subsequent release it would be breaking as the operations have moved to a new hook from `useGallery`.

## Related Issue





Closes [PWA-1672](https://jira.corp.magento.com/browse/PWA-1672).

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. In CE and EE, check that the gallery view on the home page renders the wishlist button.
2. Validate that the workflows continue to function when adding an item in all situations (only single wishlist enabled/multiple wishlist enabled, etc)
3. Verify functionality from #3105 works on the category pages.

## Screenshots / Screen Captures (if appropriate)

![](https://i.gyazo.com/546563446f8fc5137a4a2f86bda3bed3.gif)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
